bot/cogs/news.py

```diff
@@ -0,0 +1,234 @@
+
+PEPS_RSS_URL = "https://www.python.org/dev/peps/peps.rss/"
+
+RECENT_THREADS_TEMPLATE = "https://mail.python.org/archives/list/{name}@python.org/recent-threads"
+THREAD_TEMPLATE_URL = "https://mail.python.org/archives/api/list/{name}@python.org/thread/{id}/"
+MAILMAN_PROFILE_URL = "https://mail.python.org/archives/users/{id}/"
+THREAD_URL = "https://mail.python.org/archives/list/{list}@python.org/thread/{id}/"
+
+AVATAR_URL = "https://www.python.org/static/opengraph-icon-200x200.png"
+
+log = logging.getLogger(__name__)
+
+
+class News(Cog):
+    """Post new PEPs and Python News to `#python-news`."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.webhook_names = {}
+        self.webhook: t.Optional[discord.Webhook] = None
+        self.channel: t.Optional[discord.TextChannel] = None
+
+        self.bot.loop.create_task(self.get_webhook_names())
+        self.bot.loop.create_task(self.get_webhook_and_channel())
+
+    async def start_tasks(self) -> None:
+        """Start the tasks for fetching new PEPs and mailing list messages."""
+        self.fetch_new_media.start()
+
+    @loop(minutes=20)
+    async def fetch_new_media(self) -> None:
+        """Fetch new mailing list messages and then new PEPs."""
+        await self.post_maillist_news()
+        await self.post_pep_news()
+
+    async def sync_maillists(self) -> None:
+        """Sync currently in-use maillists with API."""
+        # Wait until guild is available to avoid running before everything is ready
+        await self.bot.wait_until_guild_available()
+
+        response = await self.bot.api_client.get("bot/bot-settings/news")
+        for mail in constants.PythonNews.mail_lists:
+            if mail not in response["data"]:
+                response["data"][mail] = []
+
+        # Because we are handling PEPs differently, we don't include it to mail lists
+        if "pep" not in response["data"]:
+            response["data"]["pep"] = []
+
+        await self.bot.api_client.put("bot/bot-settings/news", json=response)
+
+    async def get_webhook_names(self) -> None:
+        """Get webhook author names from maillist API."""
+        await self.bot.wait_until_guild_available()
+
+        async with self.bot.http_session.get("https://mail.python.org/archives/api/lists") as resp:
+            lists = await resp.json()
+
+        for mail in lists:
+            if mail["name"].split("@")[0] in constants.PythonNews.mail_lists:
+                self.webhook_names[mail["name"].split("@")[0]] = mail["display_name"]
+
+    async def post_pep_news(self) -> None:
+        """Fetch new PEPs and when they don't have announcement in #python-news, create it."""
+        # Wait until everything is ready and http_session available
+        await self.bot.wait_until_guild_available()
+        await self.sync_maillists()
+
+        async with self.bot.http_session.get(PEPS_RSS_URL) as resp:
+            data = feedparser.parse(await resp.text())
```

Comment:
We need to specify an encoding here to prevent errors which I've experienced locally. I think utf-8 will do the trick?
```suggestion
            data = feedparser.parse(await resp.text("utf-8"))
```